src/orchestrator.ts

```diff
@@ -185,6 +180,16 @@ export class Orchestrator {
                     context.df.isReplaying = state[partialResult.completionIndex].IsPlayed;
                 }
 
+                if (TaskFilter.isFailedTask(partialResult)) {
+                    if (!gen.throw) {

```

Comment:
When would we expect `gen.throw` to be undefined? Can you add a comment explaining this?

Reply:
I believe this is mainly to satisfy the TypeScript type-checker, as it is technically nullable I believe? Probably should never happen.

Reply:
+1, this is simply because the type of `gen` is `IterableIterator<unknown>`. That `unknown` means that we need to be quite safe about what properties we access, thus the check.

A cleaner way to go about this is probably to assign it a more concrete type (the generator type) earlier in this method. However, this snippet of code already existed and has simply been moved down, so this bit where we check "does `gen` have a `throw` method?" is not being introduced in this PR :)

Reply:
Good to know. A comment might be a nice idea since I think this will be non-obvious to other readers like it was for me.

Reply:
Done!